src/metabase/automagic_dashboards/filters.clj

```diff
@@ -0,0 +1,93 @@
+(ns metabase.automagic-dashboards.filters
+  (:require [clojure.string :as str]
+            [metabase.models.field :refer [Field]]
+            [schema.core :as s]
+            [toucan.db :as db]))
+
+(def ^:private FieldIdForm
+  [(s/one (s/constrained (s/either s/Str s/Keyword)
+                         (comp #{"field-id" "fk->"} str/lower-case name))
+          "head")
+   s/Any])
+
+(defn- collect-field-references
+  [card]
+  (->> card
+       :dataset_query
+       :query
+       ((juxt :breakout :fields))
+       (tree-seq sequential? identity)
+       (remove (s/checker FieldIdForm))
+       (map last)))
+
+(defn- candidates-for-filtering
+  [cards]
+  (->> cards
+       (mapcat collect-field-references)
+       distinct
+       (map Field)
+       (filter (fn [{:keys [base_type special_type]}]
+                 (or (isa? base_type :type/DateTime)
```

Comment:
So we don't want filtering for unix timestamps here? Any reason to exclude them?